packages/nx/src/native/tasks/task_hasher.rs

```diff
@@ -220,6 +220,13 @@ impl TaskHasher {
                 trace!(parent: &span, "hash_env: {:?}", now.elapsed());
                 hashed_env
             }
+            HashInstruction::Cwd(mode) => {
+                let workspace_root = std::path::Path::new(&self.workspace_root);
+                let cwd = std::env::current_dir().unwrap_or_default();
```

Comment:
This can't use the process cwd... because this might be on the daemon...when it should be the one from the main nx command.